app/controllers/planning_applications/review/immunity_details_controller.rb

```diff
@@ -3,32 +3,19 @@
 module PlanningApplications
   module Review
     class ImmunityDetailsController < BaseController
-      before_action :set_immunity_detail, only: %i[show edit update]
-      before_action :set_review_immunity_detail, only: %i[show edit update]
-
-      def show
-        respond_to do |format|
-          format.html
-        end
-      end
-
-      def edit
-        respond_to do |format|
-          format.html
-        end
-      end
+      before_action :set_immunity_detail, only: :update
+      before_action :set_review_immunity_detail, only: :update
```

Comment:
Why not just call these as methods inside the `update` method, since there's only one action?